src/sentry/api/endpoints/project_codeowners_details.py

```diff
@@ -0,0 +1,64 @@
+import logging
+
+from rest_framework import status
+from rest_framework.response import Response
+from rest_framework.exceptions import NotFound
+from sentry.api.bases.project import ProjectEndpoint
+from sentry.api.serializers import serialize
+from sentry.models import ProjectCodeOwners
+
+from .project_codeowners import ProjectCodeOwnerSerializer
+
+from sentry.api.endpoints.project_ownership import ProjectOwnershipMixin
+
+logger = logging.getLogger(__name__)
+
+
+class ProjectCodeOwnersDetailsEndpoint(ProjectEndpoint, ProjectOwnershipMixin):
+    def convert_args(
+        self, request, organization_slug, project_slug, codeowners_id, *args, **kwargs
+    ):
+        args, kwargs = super().convert_args(
+            request, organization_slug, project_slug, *args, **kwargs
+        )
+        try:
+            kwargs["codeowners"] = ProjectCodeOwners.objects.get(
+                id=codeowners_id, project=kwargs["project"]
+            )
+        except ProjectCodeOwners.DoesNotExist:
+            raise NotFound(detail=f"Cannot find a ProjectCodeOwners with id {codeowners_id}")
```

Comment:
Do we normally return the id of the object they are trying to find?

Reply:
Hmmm, I guess not. Is there any reason we shouldn't? I think a descriptive error message saying exactly whats wrong with the request is better developer experience. I don't think the id is privileged info, since its part of the request url.

Reply:
Well it could also be the wrong `id` _with_ that particular `project`. I think it's more accurate to just say like we couldn't find it imo

Reply:
@MeredithAnya 👍  I also found that we are mostly using the `ResourceDoesNotExist` exception for this case. So I updated  ProjectCodeOwners to stick with our current convention.